lib/utils.js

```diff
@@ -161,11 +161,20 @@ module.exports.mapToMac = function (options) {
 module.exports.actionJackerDecorator = function (emitter, options, fn, mapper) {
   options = clone(options);
   fn = fn || function (err, data) {};
-  return function (err, data) {
-    fn.apply(emitter, [err, data]);
-    if (err || !mapper || !data) return;
-
-    var key = mapper(data);
+  return function (err, data) {
+
+    var resultantData = data;

```

Comment:
It is the same but I was just keeping it a bit more immutable in a sense and not muddying up the data passed in.